Assets/Scripts/PlayerFall.cs

```diff
@@ -0,0 +1,69 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class PlayerFall : MonoBehaviour
+{
+    
+    private Rigidbody PlayerBod;
+    //speed at which the player is dragged down
+    public float thrust = 10.0f;
+    private GameObject Camera;
+    private GameObject Player;
+    //height from Player we want Camera to be
+    private float desiredCameraHeight = 5.0f;
+    private Vector3 direction;
+    private float timer = 0.0f;
+    private float delayuntilFastFall = .5f;
+    
+    //falling audio at this time overriden in BeginFall method
+    [SerializeField]
+    private AudioSource fallingAudio;
+    
+    //flag
+    public bool trapOpened {get; set;}
+    
+    void Start(){
+        
+        GameObject.Find("TrapDoor").transform.GetChild(0).transform.GetChild(0).GetComponent<TrapDoorOpen>().TrapOpened += BeginFall;
+        GameObject.Find("TrapDoor").transform.GetChild(1).transform.GetChild(0).GetComponent<TrapDoorOpen>().TrapOpened += BeginFall;
+    }
+    
+    void BeginFall(){
+        direction = new Vector3(0, desiredCameraHeight, 0);
+        Player = GameObject.Find("MainPlayer");

```

Comment:
First, for the love of god, stop using `GameObject.Find`. Use `GameObject.FindWithTag`.

Second, cache this shit in the Awake/Start method instead. These find method calls are expensive.

Reply:
Duly Noted. Don't hit me :(